### PR TITLE
Wrap up

### PR DESCRIPTION
**Pull request by n-osborne**
This PR hopefully brings the last changes to be able to merge into main.

Mainly:
- generate the last pieces of the stm module
- some last minutes fixes
- new tests
- dependencies

### CODE REVIEW

File: plugins/qcheck-stm/test/expect_stm.expected [hunk 1]
```diff
@@ -92,7 +94,10 @@ module Spec =
       | To_list -> Res ((list char), (to_list sut__010_))
   end
 module STMTests = (STM_sequential.Make)(Spec)
+<<<<<<< HEAD
```
Comment (shym) [inline plugins/qcheck-stm/test/expect_stm.expected]: This should not appear.

File: plugins/qcheck-stm/src/stm_of_ir.ml [hunk 2]
```diff
@@ -576,6 +576,7 @@ let stm config ir =
   let spec_expr =
     pmod_structure
       [
+        open_mod "STM";
```
Comment (shym) [inline plugins/qcheck-stm/src/stm_of_ir.ml:590]: The diff of that commit does not match the message, I imagine there was some mixed up somewhere.

File: plugins/qcheck-stm/test/expect_stm.expected [hunk 3]
```diff
@@ -1,6 +1,4 @@
 open Lib
-open QCheck
```
Comment (shym) [inline plugins/qcheck-stm/test/expect_stm.expected]: The changes of that commit (dbab08) do not match the commit message.

File: ortac-qcheck-stm.opam [hunk 4]
```diff
@@ -30,6 +30,9 @@ depends: [
   "ppxlib" {>= "0.26.0"}
   "gospel"
   "ortac"
+  "ortac-runtime" {with-test}
```
Comment (shym) [inline ortac-qcheck-stm.opam]: I have a little doubt there: indeed we can install our plugin and even use it without those dependencies, but the plugin is then clearly useless. As we are generating an executable command, I think putting all the dependencies that are required to actually fully use it is rather fair. (There debian would draw its nuances of recommends and suggests but I don’t think we have that).

**Review by shym — COMMENTED** [review]
I see two commits whose messages do not match the content. I think there might have been a mishap with commit 557ad08 containing a lot of code it was not meant to contain, maybe the content of those commits. `git reflog` might come to the rescue to dig up what went wrong during some rebase?

**Comment by n-osborne** [comment]
History is rewritten and should be correct :crossed_fingers:

**Review by shym — APPROVED** [review]
I’ve reviewed the range 8730457 (included)..bbb48274. Very good job!
(The only comment that came to mind is a note to our future selves that it might be interesting to generate `type nonrec` to avoid name collisions. But that’ll require some `module` magic anyway to cover most cases so let’s wait.)

**Pull request merged**